laceworksdk/http_session.py

```diff
@@ -194,141 +199,191 @@ def _get_request_headers(self, org_access=False):
+
+        # Check for 'data' or 'json'
+        data = kwargs.get("data", "")
+        json = kwargs.get("json", "")
+        if data or json:
+            logger.debug(f"{method} request data:\n{data}{json}")
+
+        # TODO: Remove this on v1.0 release - this is done for back compat
+        if data and not json:
+            kwargs["json"] = data
+            kwargs.pop("data")
+
+        # Make the HTTP request to the API endpoint
+        response = self._session.request(method, uri, headers=headers, **kwargs)
 
         # Validate the response
         self._check_response_code(response, DEFAULT_SUCCESS_RESPONSE_CODES)
 
         self._print_debug_response(response)
 
+        if response.status_code == 204:
+            try:
+                response.json()
+            except Exception:
+                response._content = b'{"data": []}'
+
         return response
 
-    def patch(self, uri, org=False, data=None, param=None):
+    def get(self, uri, params=None, **kwargs):
         """
-        :param uri: uri to send the HTTP POST request to
-        :param org: boolean representing whether the request should be performed at the Organization level
-        :param data: json object containing the data
-        :param param: python object containing the parameters
+        :param uri: uri to send the HTTP GET request to
+        :param params: dict of parameters for the HTTP request
+        :param kwargs: passed on to the requests package
 
         :return: response json
 
-        :raises: ApiError if unable to get a connection
+        :raises: ApiError if anything but expected response code is returned
         """
 
-        self._check_access_token()
+        # Perform a GET request
+        response = self._request("GET", uri, params=params, **kwargs)
 
-        uri = f"{self._base_url}{uri}"
+        return response
+
+    def get_pages(self, uri, params=None, **kwargs):
+        """
+        :param uri: uri to send the initial HTTP GET request to
+        :param params: dict of parameters for the HTTP request
+        :param kwargs: passed on to the requests package
 
-        logger.info(f"PATCH request to URI: {uri}")
-        logger.debug(f"PATCH request data:\n{data}")
+        :return: a generator that Gets and yields pages of data
 
-        # Perform a PATCH request
-        response = self._session.patch(uri, params=param, json=data, headers=self._get_request_headers(org_access=org))
+        :raises: ApiError if anything but expected response code is returned
+        """
 
-        # Validate the response
-        self._check_response_code(response, DEFAULT_SUCCESS_RESPONSE_CODES)
+        response = self.get(uri, params=params, **kwargs)
 
-        self._print_debug_response(response)
+        while True:
+            yield response
 
-        return response
+            try:
+                next_page = response.json()['paging']['urls']['nextPage']
+            except Exception:
```

Comment:
this is a pretty broad exception handling here... can y8ou narrow that down?

also better to split this up a bit

at least to...

```python
response_json = response.json()
next_page = response_json.get("paging", {}).get("urls", {}).get("nextPage")
```

That way you at least don't get KeyError there, etc. (also single vs double quote for consistency in the code base)

Reply:
you marked this as resolved, yet I still see the same code, with a `except Exception` here

(that is to say, you did adress part of the comment, just not all of it, and not the broad exception handling aspect of it)